lib/index.js

```diff
@@ -1,13 +1,19 @@
 var fs    = require('fs');
 var Joi   = require('joi');
+var glob  = require('glob');
 var path  = require('path');
 
+var internals = {};
+
 exports.register = function (server, options, next) {
 
+  internals.bookshelf = null;
+  internals.modelObjects = [];
+
   var schema = {
     knex: Joi.object().required(),
     plugins: Joi.array().includes(Joi.string()).default([]),
-    models: Joi.string().required(),
+    models: Joi.alternatives().try(Joi.string(), Joi.array()).required(),
```

Comment:
I think the better solution here would be to have `Joi.array().single().items(Joi.string()).required()`.

This would make it so the user can pass in either an array or a single item, but Joi would convert it into an array so you don't have to check it's type, you can just iterate over it. This will mean that you'll have to change the `Joi.assert` to a `Joi.validate` on line 22 as well.

This will allow for the case of a user passing in a array of non-glob directories (which you should have a test case for). Right now, your setup only allows for:

1. a single directory
2. a single glob
3. an array of globs